### PR TITLE
fix(upgd): report the unscaled prediction loss in metrics

### LINKED ISSUE #374: UPGD metrics[0] (mean_loss) is multiplied by the slow-simplex gradient budget and reads 0.0 under step2_strict_digit_readout_default, hiding task-switch error
## Summary

`UPGDUpdateResult.metrics` is documented as `[mean_loss, mean_utility, min_utility, max_perturbation_magnitude]` and is the only per-step loss `run_upgd_arrays` / `run_upgd_loop` emit. In `UPGDLearner.update` (`alberta_framework/core/upgd.py`) the two-timescale simplex readout multiplies the returned `loss_value` by `slow_simplex_gradient_scale` — both in `loss_and_aux_fn` (`loss = slow_simplex_gradient_scale * mse_loss`) and in the direct-MSE branch (`loss_value = slow_simplex_gradient_scale * 0.5 * jnp.sum(sq_masked) / denom`). That scale is `readout_slow_simplex_gradient_multiplier`, documented as a *gradient budget* knob ("allocates the bounded gradient budget to the fast simplex branch"). The shipped factory `UPGDLearner.step2_strict_digit_readout_default` sets it to `0.0`, so once the adaptive simplex gate opens on persistent simplex targets, `metrics[0]` reads exactly `0.0` regardless of how wrong the predictions are. Internal gates use `step_mse` and are unaffected, so nothing else surfaces the error.

## Falsifier (deterministic, pristine `main` @ `45279f3`)

```python
import jax.numpy as jnp, jax.random as jr
from alberta_framework.core.upgd import UPGDLearner
learner = UPGDLearner.step2_strict_digit_readout_default(n_heads=3, hidden_sizes=(8,))
state = learner.init(4, jr.key(0)); obs = jnp.ones(4)
A, B = jnp.array([1., 0., 0.]), jnp.array([0., 0., 1.])
for t in range(260):
    tgt = A if t < 200 else B                       # one task switch at t=200
    r = learner.update(state, obs, tgt); state = r.state
    if t in (199, 200, 201, 205, 220):
        print(t, "metrics[0] =", float(r.metrics[0]), " true 0.5*SE =", round(float(0.5*jnp.sum((r.predictions-tgt)**2)), 6))
```

Output on `main`:

```
199 metrics[0] = 0.0  true 0.5*SE = 4.1e-05
200 metrics[0] = 0.0  true 0.5*SE = 0.989012
201 metrics[0] = 0.0  true 0.5*SE = 0.976516
205 metrics[0] = 0.0  true 0.5*SE = 0.580911
220 metrics[0] = 0.0  true 0.5*SE = 0.006874
```

The reported loss is a flat zero across the task switch — a learning curve, plasticity, or forgetting number read from `metrics[:, 0]` for this factory erases exactly the spike a continual-learning lane exists to measure and looks like a perfect learner.

Expected: the gradient-budget scale shapes only the differentiated objective; `metrics[0]` reports the unscaled prediction loss (with the configured target weights and normalization).

## Dedup

`gh issue/pr list --search "mean_loss" / "slow_simplex" / "gradient multiplier" / "step2_strict_digit"`: nothing on this surface. Adjacent, closed: #33 (spread estimators), #46 (sigma0 reduction pin). No open PR touches `core/upgd.py`.

Fix PR follows (failing-test-first).

### PR DESCRIPTION
**Pull request by Prabhat1308**
Closes #374.

## Issue

The two-timescale simplex readout folded `slow_simplex_gradient_scale` (the slow-branch *gradient budget*, `0.0` in the shipped `step2_strict_digit_readout_default` factory) into the returned `loss_value`, so `metrics[0]` — documented as `mean_loss` — read identically `0.0` across a task switch whose true `0.5·SE` peaked at `0.99`.

## New state

`loss_and_aux_fn` still differentiates the scaled objective but returns the unscaled prediction loss in its auxiliary tuple, and the direct-MSE branch keeps the scale on the hand-computed gradients only; `metrics[0]` reports the unscaled loss (with the configured target weights and normalization). Gradients, updates, gates, and every other metric are bit-identical to before — only the reported number changes, and only in `two_timescale_simplex` mode.

Failing-test-first: `TestUpdateMetrics::test_reported_loss_is_not_scaled_by_the_slow_simplex_gradient_multiplier` fails on `main` (reported loss `0.0` at the switch) and passes here.

## Evidence

Falsifier from #374 at this head:

```text
 step   phase   metrics[0] reported   true 0.5*SE
  200       B            0.68458790      0.989012
  205       B            0.66639626      0.580911
  220       B            0.09658543      0.006874
```

(the reported value carries the factory's positive/negative target weights and normalization, so it is not numerically `0.5·SE`, but the switch spike is now visible instead of a flat zero)

Verification at `f3176ccd7b846233fc3df031066dd47f358329ae`:

```text
.venv/bin/python -m pytest tests/test_upgd.py tests/test_step2*.py -q -o addopts=""   -> 82 passed
.venv/bin/python -m ruff check .                                                     -> All checks passed!
.venv/bin/python -m mypy                                                             -> Success: no issues found in 164 source files
```

`core/upgd.py` is not a registered evidence source; no benchmark lane, seeds, or `outputs/` artifacts were touched; no `CHANGELOG.md` edit (maintainer-recorded).

CLI sessions cannot mint GitHub attachment URLs, so the run output above is inline rather than attached.

- [x] logs: inline above (focused pytest, ruff, mypy, falsifier output at the evidence head)

Provider/model/client: anthropic / claude-fable-5 / Claude Code 2.1.233 (contribute-to-asi skill revision 72e4239e).

### CODE REVIEW

**Review by lalalune — APPROVED** [review]
Independent review at exact head f3176cc: READY. The patch separates the unscaled prediction loss reported in metrics[0] from the slow-simplex gradient multiplier without changing the differentiated gradients or update path; other readout modes continue to report their original objective. Verified in a clean current-main composition with PR393: 385 focused/adjacent UPGD and Step 2 tests passed, Ruff and strict mypy passed, and diff hygiene is clean. No registered evidence source or output artifact changes.
